controllers/networkchaos/trafficcontrol/types.go

```diff
@@ -115,37 +135,40 @@ func (r *endpoint) Apply(ctx context.Context, req ctrl.Request, chaos v1alpha1.I
 		return err
 	}
 
-	err = m.Commit(ctx)
-	if err != nil {
-		// if pod is not found or not running, don't print error log and wait next time.
-		if err != podnetworkmanager.ErrPodNotFound && err != podnetworkmanager.ErrPodNotRunning {
-			r.Log.Error(err, "fail to commit")
-		}
-		return err
-	}
+	ch := m.Commit(ctx)
 
 	networkchaos.Status.Experiment.PodRecords = make([]v1alpha1.PodStatus, 0, len(pods))
-	for index, pod := range pods {
+	for keyErrorTuple := range ch {
+		err := keyErrorTuple.Err
+		if err != nil {
+			// if pod is not found or not running, don't print error log and wait next time.
```

Comment:
should we also print out one line info or warning log if the pod is not found or not running?

Reply:
I agree with you 👍

Reply:
There are some logs will be printed in the method `Commit()`.

Reply:
Yes, It seems that we handled the error twice, I think we'd better do it once, see the [logging best practise from dave cheney](https://dave.cheney.net/2015/11/05/lets-talk-about-logging).

Reply:
@Colstuwjx Fixed, I have removed redundant logs.

However, nearly all errors are finally "handled" by retrying globally... Maybe we should use `warn` or other levels to log these errors, but it could be a big project to refine error handling :crying_cat_face: